testing/e2e/e2e_withdrawal_test.go

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN "AS IS" BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package e2e_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"strconv"
+
+	beaconapi "github.com/attestantio/go-eth2-client/api"
+	"github.com/berachain/beacon-kit/execution/requests/eip7002"
+	"github.com/berachain/beacon-kit/node-api/handlers/beacon/types"
+	"github.com/berachain/beacon-kit/node-api/handlers/utils"
+	"github.com/berachain/beacon-kit/primitives/crypto"
+	beaconmath "github.com/berachain/beacon-kit/primitives/math"
+	"github.com/berachain/beacon-kit/primitives/version"
+	"github.com/berachain/beacon-kit/testing/e2e/suite"
+	e2etypes "github.com/berachain/beacon-kit/testing/e2e/suite/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	gethcore "github.com/ethereum/go-ethereum/core/types"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+const (
+	// DefaultWithdrawalAmount is the default withdrawal amount in Gwei for testing
+	DefaultWithdrawalAmount = 1_000_000_000 // 1 BERA (10^9 Gwei)
+
+	// WithdrawalTxGasLimit is the gas limit for withdrawal transactions
+	WithdrawalTxGasLimit = 500_000
+
+	// BlocksToWaitAfterWithdrawal is the number of blocks to wait after a withdrawal
+	BlocksToWaitAfterWithdrawal = 3
```

Comment:
nit: if this is related to some chain spec constant, can you reuse it instead of hard-coding values? It would help test maintenance IMO

Reply:
these are not chain spec constant. These are test-specific constants used in the E2E withdrawal test. They are used for testing purposes only.